packages/graphql-mocks/src/highlight/highlight.ts

```diff
@@ -71,7 +70,7 @@ export class Highlight {
     const schema = this.schema;
 
     // all changes are implemented with a fresh copy of data
-    const references = clone(this.references);
+    const references = [...this.references];
```

Comment:
This was the biggest slowdown. I should have known 🙈. In large schemas especially ones starting by highlight an entire schema `applyHighlighters` can be called a few times and if this has a field reference tuple `[type, field]`, it's doing deep copies over and over:
> Creates a deep copy of the source that can be used in place of the source object without retaining any references to it.

All that's needed here is a new array reference, everything else inside it is atomic and read-only.